### PR TITLE
Refactor resource

### PR DESCRIPTION
**Pull request by Vancelot11**

### CODE REVIEW

File: resources/proxytrack.rb [hunk 1]
```diff
@@ -22,17 +22,26 @@
...
     User=#{new_resource.user}
-    ExecStart=/bin/proxytrack #{new_resource.proxy_address}:#{new_resource.proxy_port} #{new_resource.icp_address}:#{new_resource.icp_port} #{new_resource.httrack_file_paths.join(' ')}
+    ExecStart=/bin/proxytrack #{new_resource.proxy_address}:#{new_resource.proxy_port} #{new_resource.icp_address}:#{new_resource.icp_port} #{new_resource.httrack_file_paths.sort.join(' ')}
 
     [Install]
     WantedBy=multi-user.target
     EOF
-    action [:create, :start, :enable]
+    action [:create]
+    notifies :restart, "service[proxytrack-#{new_resource.name}]", :immediately
```
Comment (ramereth) [inline resources/proxytrack.rb]: I don't think we need to restart this immediately.
```suggestion
    notifies :restart, "service[proxytrack-#{new_resource.name}]"
```

File: resources/proxytrack.rb [hunk 2]
```diff
@@ -22,17 +22,26 @@
...
 end
 
 action :delete do
   systemd_unit "proxytrack-#{new_resource.name}.service" do
-    action [:stop, :disable, :delete]
+    action [:delete]
+  end
+
+  service "proxytrack-#{new_resource.name}" do
+    action [:stop, :disable]
```
Comment (ramereth) [inline resources/proxytrack.rb:41]: You should probably move this above the ``systemd_unit`` resource since that deletes the unit.

**Review by ramereth — CHANGES_REQUESTED** [review]
Please also update ChefSpec

**Comment by ramereth** [comment]
[LF Archived Website Migration](https://trello.com/c/CcakKTMo/33-lf-archived-website-migration)

File: spec/proxytrack-test/resource_test_spec.rb [hunk 3]
```diff
@@ -37,6 +37,10 @@
           action: [:create]
         )
       end
+      it { expect(chef_run).to create_systemd_unit('proxytrack-test.com.service') }
+      it { expect(chef_run).to enable_service('proxytrack-test.com') }
+      it { expect(chef_run).to start_service('proxytrack-test.com') }
```
Comment (ramereth) [inline spec/proxytrack-test/resource_test_spec.rb:44]: Can you add a test for the notification please?

File: spec/proxytrack-test/resource_test_spec.rb [hunk 4]
```diff
@@ -37,6 +38,12 @@
           action: [:create]
         )
       end
+      it { expect(chef_run).to create_systemd_unit('proxytrack-test.com.service') }
+      it { expect(unit).to notify('service[proxytrack-test.com]').to(:restart) }
```
Comment (ramereth) [inline spec/proxytrack-test/resource_test_spec.rb]: I'd prefer we go with this and remove the ``let`` statement above.
```suggestion
      it { expect(chef_run.systemd_unit('proxytrack-test.com.service')).to notify('service[proxytrack-test.com]').to(:restart) }
```

**Review by ramereth — APPROVED** [review]
Please do a patch bump.

**Comment by Vancelot11** [comment]
!bump patch ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a patch-level version bump to v1.0.1.  Have a nice day!

**Pull request merged**